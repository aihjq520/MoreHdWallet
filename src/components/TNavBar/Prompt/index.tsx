import ReactDOM from 'react-dom'

const Prompt = ({
  visible = false,
  color,
  children
}: {
  visible?: boolean
  color?: 'default' | 'warn' | 'error' | 'info'
  children?: React.ReactNode
}) => {
  const colorClass = () => {
    switch (color) {
      case 'default':
        return 'bg-gray-400 text-white border-gray-500'
      case 'warn':
        return 'bg-yellow-50 text-red-400 border-red-50'
      case 'error':
        return 'bg-red-500 text-white border-white'
      case 'info':
        return 'bg-blue-200 text-blue-500 '
      default:
        return 'bg-gray-400 text-white border-gray-500'
    }
  }

  return ReactDOM.createPortal(
    <div
      className={`absolute z-max border-b transition duration-75 bottom-full w-full pt-6 pb-3 px-3 ${
        visible ? 'translate-y-full' : 'translate-y-0'
      } ${colorClass()}`}
    >
      {children}
    </div>,
    document.body
  )
}

export default Prompt
