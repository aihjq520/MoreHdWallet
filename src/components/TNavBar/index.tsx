import React from 'react'
import { useNavigate } from 'react-router-dom'

import { NavBar } from 'antd-mobile'

type ColorType = 'default' | 'theme'

const TNavBar = ({
  title,
  height = 12,
  hasBack = true,
  color = 'default',
  right
}: {
  title?: string
  height?: number
  hasBack?: boolean
  color?: ColorType
  right?: React.ReactNode
}) => {
  const navigate = useNavigate()

  const isDefaultColor = color === 'default'
  return (
    <NavBar
      style={{
        '--height': `${height * 0.25}rem`
      }}
      className={`fixed z-50 w-screen ${
        isDefaultColor ? 'bg-white' : 'bg-theme text-white'
      }`}
      backArrow={hasBack}
      onBack={() => navigate(-1)}
      right={right}
    >
      {title}
    </NavBar>
  )
}

export default TNavBar
