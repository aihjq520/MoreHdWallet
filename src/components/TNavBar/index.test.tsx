import { render } from '@testing-library/react'

import TNavBar from './index'

vi.mock('react-router-dom', () => ({
  ...vi.importActual('react-router-dom'),
  useNavigate: () => vi.fn()
}))

test('renders match snapshot', () => {
  const { asFragment } = render(<TNavBar title="测试" />)
  expect(asFragment()).matchSnapshot()
})
