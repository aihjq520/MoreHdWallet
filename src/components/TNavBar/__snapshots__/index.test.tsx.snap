// Vitest Snapshot v1

exports[`renders match snapshot 1`] = `
<DocumentFragment>
  <div
    class="adm-nav-bar fixed z-50 w-screen bg-white"
    style="--height: 45px;"
  >
    <div
      class="adm-nav-bar-left"
      role="button"
    >
      <div
        class="adm-nav-bar-back"
      >
        <span
          class="adm-nav-bar-back-arrow"
        >
          <svg
            class="antd-mobile-icon"
            height="1em"
            style="vertical-align: -0.125em;"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <title>
              A204776C-9DAE-4D2F-806A-362964924114@2x
            </title>
            <g
              fill="none"
              fill-rule="evenodd"
              id="LeftOutline-LeftOutline"
              stroke="none"
              stroke-width="1"
            >
              <g
                id="LeftOutline-编组"
              >
                <rect
                  fill="#FFFFFF"
                  height="48"
                  id="LeftOutline-矩形"
                  opacity="0"
                  width="48"
                  x="0"
                  y="0"
                />
                <path
                  d="M31.7053818,5.11219264 L13.5234393,22.6612572 L13.5234393,22.6612572 C12.969699,23.2125856 12.9371261,24.0863155 13.4257204,24.6755735 L13.5234393,24.7825775 L31.7045714,42.8834676 C31.7795345,42.9580998 31.8810078,43 31.9867879,43 L35.1135102,43 C35.3344241,43 35.5135102,42.8209139 35.5135102,42.6 C35.5135102,42.4936115 35.4711279,42.391606 35.3957362,42.316542 L16.7799842,23.7816937 L16.7799842,23.7816937 L35.3764658,5.6866816 C35.5347957,5.53262122 35.5382568,5.27937888 35.3841964,5.121049 C35.3088921,5.04365775 35.205497,5 35.0975148,5 L31.9831711,5 C31.8795372,5 31.7799483,5.04022164 31.7053818,5.11219264 Z"
                  fill="currentColor"
                  fill-rule="nonzero"
                  id="LeftOutline-路径"
                />
              </g>
            </g>
          </svg>
        </span>
        <span
          aria-hidden="true"
        />
      </div>
    </div>
    <div
      class="adm-nav-bar-title"
    >
      测试
    </div>
    <div
      class="adm-nav-bar-right"
    />
  </div>
</DocumentFragment>
`;
